core/lib/types/src/event.rs

```diff
@@ -51,6 +53,61 @@ static L1_MESSAGE_EVENT_SIGNATURE: Lazy<H256> = Lazy::new(|| {
     )
 });
 
+// struct L2ToL1Log {
+//     uint8 l2ShardId;
+//     bool isService;
+//     uint16 txNumberInBlock;
+//     address sender;
+//     bytes32 key;
+//     bytes32 value;
+// }
+#[derive(Debug, Default, Clone)]
+pub struct L1MessengerL2ToL1Log {
+    l2_shard_id: u8,
+    is_service: bool,
+    tx_number_in_block: u16,
+    sender: Address,
+    key: U256,
+    value: U256,
+}
+
+impl L1MessengerL2ToL1Log {
+    pub fn packed_encoding(&self) -> Vec<u8> {
+        let mut res = vec![];
+        res.extend_from_slice(&self.l2_shard_id.to_be_bytes());
+        res.extend_from_slice(&(self.is_service as u8).to_be_bytes());
```

Comment:
Bikeshedding: `u8::to_be_bytes()` looks weird; you could just `push()` a `u8` value.